core/predicate_check_test.go

```diff
@@ -19,446 +21,325 @@ import (
-				ctrl := gomock.NewController(t)
-				predicate1 := precompileconfig.NewMockPredicater(ctrl)
-				arg1 := predicate.Predicate{{1}}
-				predicate1.EXPECT().PredicateGas(arg1).Return(uint64(0), nil).Times(2)
-				predicate1.EXPECT().VerifyPredicate(gomock.Any(), arg1).Return(nil)
-				predicate2 := precompileconfig.NewMockPredicater(ctrl)
-				arg2 := predicate.Predicate{{2}}
-				predicate2.EXPECT().PredicateGas(arg2).Return(uint64(0), nil).Times(2)
-				predicate2.EXPECT().VerifyPredicate(gomock.Any(), arg2).Return(testErr)
-				return map[common.Address]precompileconfig.Predicater{
-					addr1: predicate1,
-					addr2: predicate2,
-				}
+		{
+			name: "one_tx_one_address_one_invalid_predicate",
+			txs: []*types.Transaction{
+				invalidTx,
 			},
-			accessList: types.AccessList([]types.AccessTuple{
-				{
-					Address: addr1,
-					StorageKeys: []common.Hash{
-						{1},
-					},
-				},
-				{
-					Address: addr2,
-					StorageKeys: []common.Hash{
-						{2},
-					},
+			expected: predicate.BlockResults{
+				invalidTx.Hash(): {
+					addr: set.NewBits(0),
 				},
-			}),
-			expectedRes: predicate.PrecompileResults{
-				addr1: set.NewBits(),  // valid bytes
-				addr2: set.NewBits(0), // invalid bytes
 			},
-			expectedErr: nil,
 		},
-		"two predicates niether named by access list": {
-			gas:              61600,
-			predicateContext: predicateContext,
-			createPredicates: func(t testing.TB) map[common.Address]precompileconfig.Predicater {
-				predicater := precompileconfig.NewMockPredicater(gomock.NewController(t))
-				return map[common.Address]precompileconfig.Predicater{
-					addr1: predicater,
-					addr2: predicater,
-				}
+		{
+			name: "multiple_txs",
+			txs: []*types.Transaction{
+				validTx,
+				invalidTx,
 			},
-			accessList: types.AccessList([]types.AccessTuple{
-				{
-					Address: addr3,
-					StorageKeys: []common.Hash{
-						{1},
-					},
+			expected: predicate.BlockResults{
+				validTx.Hash(): {
+					addr: set.NewBits(),
 				},
-				{
-					Address: addr4,
-					StorageKeys: []common.Hash{
-						{1},
-					},
+				invalidTx.Hash(): {
+					addr: set.NewBits(0),
 				},
-			}),
-			expectedRes: make(predicate.PrecompileResults),
-			expectedErr: nil,
-		},
-		"insufficient gas": {
-			gas:              53000,
-			predicateContext: predicateContext,
-			createPredicates: func(t testing.TB) map[common.Address]precompileconfig.Predicater {
-				predicater := precompileconfig.NewMockPredicater(gomock.NewController(t))
-				arg := predicate.Predicate{{1}}
-				predicater.EXPECT().PredicateGas(arg).Return(uint64(1), nil)
-				return map[common.Address]precompileconfig.Predicater{
-					addr1: predicater,
-				}
 			},
-			accessList: types.AccessList([]types.AccessTuple{
-				{
-					Address: addr1,
-					StorageKeys: []common.Hash{
-						{1},
-					},
-				},
-			}),
-			expectedErr: ErrIntrinsicGas,
 		},
-	} {
-		t.Run(name, func(t *testing.T) {
+	}
```

Comment:
Would a test case for a tx with no predicates make sense? (key present with nil vs omitted)